src/utilities/hypre_memory.c

```diff
@@ -12,7 +12,13 @@
  *****************************************************************************/
 
 #include "_hypre_utilities.h"
+#include "hypre_cuda_utils.h"
```

Comment:
Our header file stuff just got a lot more complicated.  Are all of these include files really needed?  I would like to maintain an overall strategy for header file use in the code (I realize there are inconsistencies already, but we should be trying to fix those).  Also, the directory-structure-dependent includes are something we should be avoiding.

Reply:
This file is very involving; it covers our memory model with all kinds of scenarios. Previously, these headers were hidden in `_hypre_utilities.h` but they had to be moved out now.